Makefile

```diff
@@ -38,6 +38,7 @@ ifeq (0,${WAYLAND})
 SRC := $(sort $(shell ${FIND} src/ -not \( -path src/wayland -prune \) -name '*.c'))
 else
 # with Wayland support
+CFLAGS += -DHAVE_WL_CURSOR_SHAPE -DHAVE_WL_EXT_IDLE_NOTIFY
```

Comment:
Could you explain why you added separate flags for these protocols?

Reply:
A bit of a pedantry, a bit of a preparation for the meson PR.

It is highly recommended to regenerate the protocol sources from xml at the build time, as the `wayland-scanner` output changes and may start relying on a new APIs (like the [`wl_proxy_marshal_flags`](https://gitlab.freedesktop.org/wayland/wayland/-/merge_requests/162) introduction in 1.19.91).

Doing that the way this repo is organized currently, with the pre-built protocol sources, will result in breaking compatibility with libwayland 1.19 (and it was a pain to get an old enough `wayland-scanner` to build the files for this PR).
On the other hand, building all the protocols every time would require w-p >= 1.32, which is a bit too hard on the users. The middle ground is to make support for the newer protocols conditional on the w-p version available during build.

I'm honestly giving up on solving this via Makefile (nobody runs `make wayland-protocols` at their system/during packaging, right?) and laying the groundwork for a proper conditionals in meson when #1226 lands. E.g.
```meson
if wayland_protos.version() >= '1.27'
     client_protocols += wl_protocol_dir / 'staging/ext-idle-notify/ext-idle-notify-v1.xml'
     add_project_arguments('-DHAVE_WL_EXT_IDLE_NOTIFY', language: 'c')
endif

if wayland_protos.version() >= '1.32'
    client_protocols += [
        wl_protocol_dir / 'staging/cursor-shape/cursor-shape-v1.xml',
        wl_protocol_dir / 'unstable/tablet/tablet-unstable-v2.xml',
    ]
    add_project_arguments('-DHAVE_WL_CURSOR_SHAPE', language: 'c')
endif
```
And sorry if it looks weird out of context.

Reply:
Okay, makes sense. Thanks for the explanation.